### PR TITLE
fix #6582 comprehensive fix to BrokenPipeError

### LINKED ISSUE #6582: More BrokenPipeError
```
error: BrokenPipeError(32, 'Broken pipe')
command: /home/ubuntu/anaconda3/envs/tensorflow_p36/bin/conda install -p /home/ubuntu/anaconda3/envs/tensorflow_p36 tensorflow-gpu -y
user_agent: conda/4.4.4 requests/2.18.4 CPython/3.6.3 Linux/4.4.0-1044-aws ubuntu/16.04 glibc/2.23
_messageid: -9223372036845075603
_messagetime: 1514219033248 / 2017-12-25 10:23:53

Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/exceptions.py", line 724, in __call__
    return func(*args, **kwargs)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/cli/main.py", line 78, in _main
    exit_code = do_call(args, p)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/cli/conda_argparse.py", line 75, in do_call
    exit_code = getattr(module, func_name)(args, parser)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/cli/main_install.py", line 11, in execute
    install(args, parser, 'install')
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/cli/install.py", line 255, in install
    handle_txn(progressive_fetch_extract, unlink_link_transaction, prefix, args, newenv)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/cli/install.py", line 281, in handle_txn
    progressive_fetch_extract.execute()
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/core/package_cache.py", line 575, in execute
    exc = self._execute_actions(prec_or_spec, prec_actions)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/core/package_cache.py", line 590, in _execute_actions
    progress_bar = ProgressBar(desc, not context.verbosity and not context.quiet, context.json)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/common/io.py", line 365, in __init__
    self.pbar = tqdm(desc=description, bar_format=bar_format, ascii=True, total=1)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/_vendor/tqdm/_tqdm.py", line 767, in __init__
    self.postfix, unit_divisor))
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/_vendor/tqdm/_tqdm.py", line 201, in print_status
    fp_write('\r' + s + (' ' * max(last_len[0] - len_s, 0)))
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/conda/_vendor/tqdm/_tqdm.py", line 195, in fp_write
    fp_flush()
BrokenPipeError: [Errno 32] Broken pipe

```

### PR DESCRIPTION
**Pull request by kalefranz**
fix #6582 

Previous related PR #6540

### CODE REVIEW

**Comment by kalefranz** [comment]
Let's hold off on this PR until we hear back from Antoine.  It's not critical to get out today.

File: conda/exceptions.py [hunk 1]
```diff
@@ -757,6 +760,11 @@ def handle_exception(self, exc_val, exc_tb):
             return 1
         elif isinstance(exc_val, SystemExit):
             return exc_val.code
+        elif (isinstance(exc_val, (IOError, OSError))
```
Comment (pitrou) [inline conda/exceptions.py]: You can spell this `EnvironmentError`.

File: conda/exceptions.py [hunk 2]
```diff
@@ -757,6 +760,11 @@ def handle_exception(self, exc_val, exc_tb):
...
+              and getattr(exc_val, 'errno', None) in (EPIPE, ESHUTDOWN)):
+            # BrokenPipeError is caused by stdout or stderr being closed by a downstream program.
+            sys.stderr.write("\nBroken pipe.\n")
```
Comment (pitrou) [inline conda/exceptions.py]: You probably want to catch other errors here, if stderr was shutdown.

File: conda/common/io.py [hunk 3]
```diff
@@ -327,17 +326,11 @@ def spinner(message=None, enabled=True, json=False):
...
-                    else:
-                        sys.stdout.write("done\n")
-                    sys.stdout.flush()
-                except (IOError, OSError) as e:
-                    # Ignore BrokenPipeError and errors related to stdout or stderr being
-                    # closed by a downstream program.
-                    if e.errno not in (EPIPE, ESHUTDOWN):
-                        raise
+                if exception_raised:
+                    sys.stdout.write("failed\n")
```
Comment (pitrou) [inline conda/common/io.py]: If you got an error writing something, there's no need trying to write again.

**Comment by kalefranz** [comment]
Per @pitrou's comments here, as well as other discussion in flowdock, I think we want to basically swallow these errors as close to their source in the conda code as possible, and let whatever conda operation is in process proceed.

So that means I'll close this PR and continue down the original path started in #6540 with a new PR.  (Also taking into account some useful comments in this PR.)

File: conda/exceptions.py (hunk 1 above)
Comment (kalefranz) [reply conda/exceptions.py]: Thanks! I didn't realize `EnvironmentError` was the python 2/3 agnostic way to do this.

**Pull request closed without merging**